### PR TITLE
Add shared foreman actions release workflow

### PR DESCRIPTION
**Pull request by dosas**
https://github.com/theforeman/actions?tab=readme-ov-file#gem-release

### CODE REVIEW

File: .github/workflows/release_gem.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      - '**'
+
+jobs:
+  release:
+    name: Release gem
+    uses: theforeman/actions/.github/workflows/release-gem.yml@v0
+    with:
+      allowed_owner: theforeman
+    secrets:
+      api_key: ${{ secrets.RUBYGEM_API_KEY }}
```
Comment (ezr-ondrej) [inline .github/workflows/release_gem.yml:16]: Pardon I'm just unsure here: where is this secret coming from? Is that organizational secret? :)

Comment (evgeni) [reply .github/workflows/release_gem.yml:16]: It is not (we have per-repo secrets that can only push that one gem)
I'll create you one!

Comment (evgeni) [reply .github/workflows/release_gem.yml:16]: done

File: .github/workflows/release_gem.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+---
+name: Release
+
+on:  # yamllint disable-line rule:truthy
+  push:
+    tags:
+      - 'v*'
```
Comment (evgeni) [inline .github/workflows/release_gem.yml:7]: the tags in this repo do not use a `v` prefix: https://github.com/theforeman/ldap_fluff/tags

**Review by evgeni — CHANGES_REQUESTED** [review]

File: .github/workflows/release_gem.yml (hunk 1 above)
Comment (dosas) [reply .github/workflows/release_gem.yml:16]: Thanks!

**Review by evgeni — APPROVED** [review]

File: .github/workflows/release_gem.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+---
+name: Release
+
+on:  # yamllint disable-line rule:truthy
+  push:
+    tags:
+      - '**'
```
Comment (ezr-ondrej) [inline .github/workflows/release_gem.yml:7]: Nit: This still could be just a single `'*'`

**Review by ezr-ondrej — APPROVED** [review]
Nice, thanks! :)
@evgeni would you do the honours?

File: .github/workflows/release_gem.yml (hunk 3 above)
Comment (dosas) [reply .github/workflows/release_gem.yml:7]: I would stick to the recommendation of the [actions README](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#patterns-to-match-branches-and-tags)

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#patterns-to-match-branches-and-tags

Comment (ezr-ondrej) [reply .github/workflows/release_gem.yml:7]: Ah, didn't realize the difference :) I doubt we want tags with `/` in a name released, but sure, I'm fine with it! :+1:

**Pull request merged**